src/Comms.cpp

```diff
@@ -22,9 +41,46 @@ namespace Comms {
     }
 
     void emitPacket(Packet *packet) {
-        Serial.print(packet->data[0]);
-        Serial.print(packet->data[1]);
-        Serial.print(packet->data[2]);
-        Serial.println(packet->data[3]);
+        //generate checksum from data and add it to the packet
+        uint16_t checksum = fletcher16(packet->data, packet->len);
```

Comment:
The checksum should include the id and the len variables in the packet. Might wanna change the fletcher16 function to take in a packet pointer instead of just a byte array